app/routes/__auth/user.server.ts

```diff
@@ -14,8 +17,64 @@ export async function getUser({ headers }: Request) {
   const email = session.get('email') as string
   const groups = session.get('groups') as string[]
   const idp = session.get('idp') as string | null
-
+  const refreshToken = session.get('refreshToken') as string
+  const cognitoUserName = session.get('cognitoUserName') as string
+  const accessToken = session.get('accessToken') as string
   if (!sub) return null
+  const user = {
+    sub,
+    email,
+    groups,
+    idp,
+    refreshToken,
+    cognitoUserName,
+    accessToken,
+  }
+  if (!accessToken) await refreshUser(user)
+
+  const tokenSet = new TokenSet(
+    JSON.parse(Buffer.from(accessToken.split('.')[1], 'base64').toString())
+  )
+  if (tokenSet.expired()) await refreshUser(user)
```

Comment:
Please parse this using jose.

Reply:
Do you have docs or a reference for how to do this? I came across this, https://github.com/panva/jose/blob/cfc63778b133b8c6731080a033c7701dfd9cc9ca/docs/functions/jwe_general_decrypt.generalDecrypt.md#readme is this what we want? 

If so, where do i get the key parameter?

Reply:
I don't know why I didn't find this sooner, https://github.com/panva/jose/blob/cfc63778b133b8c6731080a033c7701dfd9cc9ca/docs/functions/util_decode_jwt.decodeJwt.md#readme